### PR TITLE
wayland.windowManager.dwl: add module

### LINKED ISSUE #9878: [New Module Request] dwl - dwm for Wayland
### Program/Service Name

dwl

### Upstream Repository

https://codeberg.org/dwl/dwl

### Configuration Documentation

https://codeberg.org/dwl/dwl/src/branch/main/README.md#configuration

### Reason for Addition

dwl is a minimal and lightweight Wayland compositor. Without a dedicated module, users must manually configure variables and session targets to run it properly.

Since Home Manager already has infrastructure for managing graphical sessions, providing this module eliminates manual setup and allows users to run dwl simply by enabling it.

### PR DESCRIPTION
**Pull request by glmlm**
### Description

Closes #9878

As a first step, this PR focuses on the systemd setup for startup.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `nix fmt` or
      `nix-shell -A dev --run treefmt`.

- [x] Code tested through `nix build .#test-all`
      or a targeted `nix run .#tests -- <pattern>`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

  ```
  {component}: {description}

  {long description}
  ```

  See [CONTRIBUTING](https://nix-community.github.io/home-manager/#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/a51598236f23c89e59ee77eb8e0614358b0e896c/modules/programs/lesspipe.nix#L11).
  - [ ] Generate a news entry. See [News](https://nix-community.github.io/home-manager/index.xhtml#sec-news)
  - [x] Basic tests added. See [Tests](https://nix-community.github.io/home-manager/index.xhtml#sec-tests)

- If this PR adds an exciting new feature or contains a breaking change.
  - [ ] Generate a news entry. See [News](https://nix-community.github.io/home-manager/index.xhtml#sec-news)

### CODE REVIEW

File: modules/services/window-managers/dwl.nix [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    home.packages =
+      let
+        systemdVariables = concatStringsSep " " cfg.systemd.variables;
+
+        systemdActivation = pkgs.writeShellScript "dwl-startup" ''
+          ${pkgs.dbus}/bin/dbus-update-activation-environment --systemd ${systemdVariables}
+          ${pkgs.systemd}/bin/systemctl --user import-environment ${systemdVariables}
+          ${pkgs.systemd}/bin/systemctl --user reset-failed
+          ${pkgs.systemd}/bin/systemctl --user start dwl-session.target
+        '';
```
Comment (khaneliman) [inline modules/services/window-managers/dwl.nix]: issue (blocking): import only the configured environment names

Argumentless `systemctl import-environment` copies the complete shell environment. The D-Bus helper already updates both D-Bus and systemd when `--systemd` is present.

```suggestion
      let
        systemdVariables = lib.escapeShellArgs cfg.systemd.variables;

        systemdActivation = pkgs.writeShellScript "dwl-startup" ''
          ${lib.optionalString (cfg.systemd.variables != [ ]) ''
            ${pkgs.dbus}/bin/dbus-update-activation-environment --systemd ${systemdVariables}
          ''}
          ${pkgs.systemd}/bin/systemctl --user reset-failed
          ${pkgs.systemd}/bin/systemctl --user start dwl-session.target
        '';
```

Cover the empty list and a variable name requiring shell escaping.

File: modules/services/window-managers/dwl.nix [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+        dwlWrapped = pkgs.writeShellScriptBin "dwl" ''
+          trap "${pkgs.systemd}/bin/systemctl --user stop dwl-session.target" EXIT
+          ${cfg.package}/bin/dwl -s ${systemdActivation} "$@"
+        '';
+      in
+      mkIf (cfg.package != null) (if cfg.systemd.enable then [ dwlWrapped ] else [ cfg.package ]);
```
Comment (khaneliman) [inline modules/services/window-managers/dwl.nix]: issue (blocking): retain dwl's non-binary package outputs

Installing only `writeShellScriptBin` drops the package's desktop file and man pages. Keep the package and give the wrapper higher collision priority.

```suggestion
      mkIf (cfg.package != null) (
        if cfg.systemd.enable then
          [
            cfg.package
            (lib.hiPrio dwlWrapped)
          ]
        else
          [ cfg.package ]
      );
```

Assert that the resulting profile contains the wrapper, desktop file, and man page.

File: modules/services/window-managers/dwl.nix [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+    home.packages =
+      let
+        systemdVariables = concatStringsSep " " cfg.systemd.variables;
+
+        systemdActivation = pkgs.writeShellScript "dwl-startup" ''
+          ${pkgs.dbus}/bin/dbus-update-activation-environment --systemd ${systemdVariables}
+          ${pkgs.systemd}/bin/systemctl --user import-environment ${systemdVariables}
+          ${pkgs.systemd}/bin/systemctl --user reset-failed
+          ${pkgs.systemd}/bin/systemctl --user start dwl-session.target
```
Comment (khaneliman) [inline modules/services/window-managers/dwl.nix]: issue (blocking): Import only configured environment names

With `systemd.variables = [ ]`, argumentless `systemctl --user import-environment` imports the complete launcher environment into the user manager. Guard both import commands when the list is empty and render configured names with `lib.escapeShellArgs`. Keep the explicit `systemctl import-environment` call because it is the `dbus-broker` fallback. Add startup-script tests for an empty list and a non-empty list.

File: modules/services/window-managers/dwl.nix [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    systemd = {
+      enable = mkEnableOption "systemd" // {
+        default = true;
+        description = ''
+          Whether to enable {file}`dwl-session.target` on dwl startup.
+          This links to {file}`graphical-session.target`.
+        '';
+      };
+
+      variables = mkOption {
```
Comment (khaneliman) [inline modules/services/window-managers/dwl.nix]: issue (blocking): Seed the Wayland session variables by default

The ordinary `enable = true` path leaves `systemd.variables` empty, so D-Bus-activated clients do not receive `WAYLAND_DISPLAY` or the other compositor session variables. This leaves the manual setup that issue #9878 asks the module to remove. Use the established River defaults for `DISPLAY`, `WAYLAND_DISPLAY`, `XDG_CURRENT_DESKTOP`, `NIXOS_OZONE_WL`, `XCURSOR_THEME`, and `XCURSOR_SIZE`, then test the generated startup script without overriding this option.

File: modules/services/window-managers/dwl.nix (hunk 2 above)
Comment (khaneliman) [inline modules/services/window-managers/dwl.nix]: issue (blocking): Keep the configured package's non-binary outputs

When systemd integration is enabled, `home.packages` contains only the generated wrapper. This drops the configured package's Wayland session desktop file and prevents its separate man output from entering the profile when man pages are enabled. Install `cfg.package` as well and give the wrapper higher collision priority, then assert that the profile contains the wrapper, desktop file, and man page.
